lighthouse-core/gather/gatherers/dobetterweb/optimized-images.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ /**
+  * @fileoverview Determines optimized jpeg/webp filesizes for all same-origin and dataURI images.
```

Comment:
Yeah my main concern was not processing CDN images at all which is most people who make some effort. The issue I ran into was canvas not dealing with them and adding `crossorigin='anonymous'` didn't necessarily fix. Fine to investigate in future PR?